### PR TITLE
feat: GitHub action for sprocket check and lint

### PR DESCRIPTION
**Pull request by adthrasher**
Add a GitHub Action to wrap Sprocket and enable easy use of the `check` and `lint` options in GitHub CI/CD workflows.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM rust:1.81
+WORKDIR /app
+
+COPY . .
```
Comment (adthrasher) [inline Dockerfile:4]: I'm not 100% certain that I need to copy anything other than the `entrypoint.sh` script into the container.

File: Dockerfile
Comment (adthrasher) [inline Dockerfile:1]: One question I have is: should Sprocket provide a Docker image generally or is it fine to just have one as part of this action?

File: LICENSE-APACHE
Comment (a-frantz) [inline LICENSE-APACHE:1]: Isn't Sprocket just MIT licensed? Why do we have the Apache license as well?

Comment (adthrasher) [reply LICENSE-APACHE:1]: Sprocket has both MIT and Apache 2.0

**Comment by adthrasher** [comment]
> LGTM, but is there a way for me to actually run it on an example repo?

I believe once it is merged into `main` that we can point to it from a workflow. I'm still digging into the docs, but that's my impression so far.

File: LICENSE-APACHE
Comment (a-frantz) [reply LICENSE-APACHE:1]: Ah I was talking from memory, I just checked and the sprocket README says `This project is licensed as either [Apache 2.0](https://github.com/stjude-rust-labs/sprocket/blob/main/LICENSE-APACHE) or [MIT](https://github.com/stjude-rust-labs/sprocket/blob/main/LICENSE-MIT) at your discretion.`

I guess this is fine then.

**Comment by a-frantz** [comment]
Shouldn't it have a version declared somewhere?

**Comment by adthrasher** [comment]
> Shouldn't it have a version declared somewhere?

Versions get managed through branches and tags (https://docs.github.com/en/actions/sharing-automations/creating-actions/about-custom-actions#using-tags-for-release-management. Something like `release/v1` as a branch would enable `action@v1` and then you'd tag specific versions off of that branch (e.g. `1.0.0`, `1.1.0`, etc.).

**Review by a-frantz — APPROVED** [review]

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  lint:
+    description: "Whether to lint the WDL document"
+    required: false
+    default: "false"
+outputs:
+  status:
+    description: "The status of the check"
+runs:
+  using: "docker"
+  image: "Dockerfile"
```
Comment (adthrasher) [inline action.yml:14]: This causes the action to build from the Dockerfile on each run, which is a bit slow given sprocket's install time.

Comment (a-frantz) [reply action.yml:14]: But it also ensures that the latest `sprocket` is run each time, right? The potential downside with a caching solution is missing updates

Comment (adthrasher) [reply action.yml:18]: Yes, since I don't pin the spocket version in Dockerfile.

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    arg=$1
+    case $1 in
+        -l|--lint) lint="--lint"; shift;;
+        -w|--deny-warnings) warnings="--deny-warnings"; shift;;
+        -n|--deny-notes) notes="--deny-notes"; shift;;
+        *) break;;
+    esac
+done
+
+exclusions=$1
```
Comment (adthrasher) [inline entrypoint.sh:17]: @a-frantz - OK, as much as I hate parsing args in bash, this should do it. Are there any other Sprocket arguments at the moment that I should support. I didn't see anything else.

Comment (a-frantz) [reply entrypoint.sh:17]: No I think this should do it!

**Review by a-frantz — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (peterhuene) [reply Dockerfile:4]: Agree that you probably only need `entrypoint.sh`

**Review by peterhuene — COMMENTED** [review]
👍

**Review by peterhuene — APPROVED** [review]
Whoops, meant for that last review to be an approval!

**Pull request merged**